pkg/image/apis/image/helper.go

```diff
@@ -381,29 +390,41 @@ func DockerImageReferenceForStream(stream *ImageStream) (DockerImageReference, e
 }
 
 // FollowTagReference walks through the defined tags on a stream, following any referential tags in the stream.
-// Will return ok if the tag is valid, multiple if the tag had at least reference, and ref and finalTag will be the last
-// tag seen. If a circular loop is found ok will be false.
-func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *TagReference, ok bool, multiple bool) {
+// Will return multiple if the tag had at least reference, and ref and finalTag will be the last tag seen.
+// If an invalid reference is found, err will be returned.
+func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *TagReference, multiple bool, err error) {
```

Comment:
Please add unit tests that exercise the new and old functionality here to prove it is correct.